### PR TITLE
fix(scrollviewer): raise PointerEnter/Exit when scrolling causes the pointer to be over a new item

### LINKED ISSUE #8938: [Skia] `ListView` pointer hover and scrolling with mouse wheel
### Current behavior

On linux platform, when hovering over an item, first it is indicated properly.
However, using the mouse wheel and scrolling down, the hover remains on that very item, even if it scrolled out of view.
Scrolling more, the hover comes into view on a fully different item.
Moving the mouse over a not hovered item, it also becomes hovered.
Scrolling up, the effect repeats. With more mouse moving, more false hovered items can be produced.
Screenshot_lvhov1
Screenshot_lvhov2
Screenshot_lvhov3
Screenshot_lvhov4
Screenshot_lvhov5

### Expected behavior

The hover indication should follow when scrolling with mouse wheel

### How to reproduce it (as minimally and precisely as possible)

Full source at link:

[uno-lvtest-master.zip](https://github.com/unoplatform/uno/files/8805102/uno-lvtest-master.zip)

### Workaround

Hovering over and out of a previously hovered item clears the hover indication.

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Uno.WinUI.Skia.Gtk Version="4.3.6"

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio Code

### IDE version

1.67.2

### Relevant plugins

Uno Platform 0.4.0

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #8938

## PR Type

What kind of change does this PR introduce?

- Bugfix

### <samp>🤖 Generated by Copilot at 2fd9da2</samp>

Enhance pointer events handling in Uno by improving the hit testing logic for wheel events. Use a `staleBranch` parameter in `HitTest` to track and update the pointer over state of UI elements.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]

**Comment by MartinZikmund** [comment]
This seems ok to me, but let's let @dr1rrb chime in

**Review by dr1rrb — COMMENTED** [review]
Sounds good for me to, however I think that we can unit test this change. You can take example on what I did for the move : https://github.com/unoplatform/uno/pull/11966/files#diff-6f227a6e07013237a705713a296838b71fcb5f73cabc7f2ee7a67ec648d9ae99R24

File: src/Uno.UI/UI/Xaml/Internal/InputManager.Pointers.Managed.cs [hunk 1]
```diff
@@ -129,8 +129,26 @@ private void OnPointerWheelChanged(Windows.UI.Core.PointerEventArgs args)
 
 			var routedArgs = new PointerRoutedEventArgs(args, originalSource);
 
-			// Second raise the event, either on the OriginalSource or on the capture owners if any
+			// First raise the event, either on the OriginalSource or on the capture owners if any
 			RaiseUsingCaptures(Wheel, originalSource, routedArgs);
+
+			// Scrolling can change the element underneath the pointer, so we need to update
+			(originalSource, var staleBranch) = HitTest(args, caller: "OnPointerWheelChanged_post_leave", isStale: _isOver);
```
Comment (dr1rrb) [inline src/Uno.UI/UI/Xaml/Internal/InputManager.Pointers.Managed.cs:136]:
```suggestion
			(originalSource, var staleBranch) = HitTest(args, caller: "OnPointerWheelChanged_post_wheel", isStale: _isOver);
```

:)

**Review by dr1rrb — APPROVED** [review]

**Comment by MartinZikmund** [comment]
@mergifyio backport legacy/4x

**Pull request merged**